homeassistant/components/livisi/coordinator.py

```diff
@@ -0,0 +1,135 @@
+"""Code to manage fetching LIVISI data API."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from aiolivisi import AioLivisi, LivisiEvent, Websocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    AVATAR_PORT,
+    CLASSIC_PORT,
+    DEVICE_POLLING_DELAY,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+)
+
+
+class LivisiDataUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Class to manage fetching LIVISI data API."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, aiolivisi: AioLivisi
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="Livisi devices",
+            update_interval=timedelta(seconds=DEVICE_POLLING_DELAY),
+        )
+        self.config_entry = config_entry
+        self.hass = hass
+        self.aiolivisi = aiolivisi
+        self.websocket = Websocket(aiolivisi)
+        self.devices: list[str] = []
```

Comment:
Make this a set. A set has unique items.

Reply:
I made this a set, thank you.